### PR TITLE
Support for variadic arguments in scalar UDFs in the C API

### PR DESCRIPTION
**Pull request by taniabogatsch**

### CODE REVIEW

File: src/main/capi/scalar_function-c.cpp [hunk 1]
```diff
@@ -45,6 +45,11 @@ duckdb::ScalarFunction &GetCScalarFunction(duckdb_scalar_function function) {
 unique_ptr<FunctionData> BindCAPIScalarFunction(ClientContext &context, ScalarFunction &bound_function,
                                                 vector<unique_ptr<Expression>> &arguments) {
 	auto &info = bound_function.function_info->Cast<CScalarFunctionInfo>();
+	if (bound_function.HasVarArgs()) {
+		for (const auto &expr : arguments) {
+			bound_function.arguments.push_back(expr->return_type);
```
Comment (Mytherin) [inline src/main/capi/scalar_function-c.cpp:50]: Could you expand on the code here? `varargs` explicitly has a type (i.e. it is not just true or false) - it looks to me that in this manner that type is ignored and is essentially set to `LogicalType::ANY`.

File: test/api/capi/capi_scalar_functions.cpp [hunk 2]
```diff
@@ -182,3 +203,51 @@ TEST_CASE("Test Scalar Functions - strings & extra_info", "[capi]") {
...
+
+	result = tester.Query("SELECT my_addition(40, 2, 100, 3)");
+	REQUIRE_NO_FAIL(*result);
+	REQUIRE(result->Fetch<int64_t>(0, 0) == 145);
+
+	result = tester.Query("SELECT my_addition(40, 42, NULL)");
+	REQUIRE_NO_FAIL(*result);
+	REQUIRE(result->IsNull(0, 0));
+
+	result = tester.Query("SELECT my_addition(NULL, 2)");
```
Comment (Mytherin) [inline test/api/capi/capi_scalar_functions.cpp:246]: Can we add a test with mismatching types (e.g. `my_addition('hello', 'world')`)?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one comment:

File: src/main/capi/scalar_function-c.cpp [hunk 3]
```diff
@@ -42,9 +42,12 @@ duckdb::ScalarFunction &GetCScalarFunction(duckdb_scalar_function function) {
 	return *reinterpret_cast<duckdb::ScalarFunction *>(function);
 }
 
-unique_ptr<FunctionData> BindCAPIScalarFunction(ClientContext &context, ScalarFunction &bound_function,
+unique_ptr<FunctionData> BindCAPIScalarFunction(ClientContext &, ScalarFunction &bound_function,
                                                 vector<unique_ptr<Expression>> &arguments) {
 	auto &info = bound_function.function_info->Cast<CScalarFunctionInfo>();
+	if (bound_function.HasVarArgs()) {
+		bound_function.arguments.assign(arguments.size(), bound_function.varargs);
```
Comment (Mytherin) [inline src/main/capi/scalar_function-c.cpp:49]: I'm still confused why this is necessary - other varargs functions don't need this, shouldn't the function overloading handle this?

Comment (taniabogatsch) [reply src/main/capi/scalar_function-c.cpp]: Yes, I wasn't aware. I removed it. 👍

**Pull request merged**